### PR TITLE
Species IP web service

### PR DESCRIPTION
**Pull request by charvolant**
Web service available via $BASE_URL/regionType/regionName/patents

eg. http://localhost:8080/oznome-demo/Indigenous+protected+areas/Angas+Downs/patents

The summary=true parameter will only return numbers

eg. http://localhost:8080/oznome-demo/Indigenous+protected+areas/Angas+Downs/patents?summary=true

Fix patent service to reuse mongodb connection
Allow metadata service to return all species

### CODE REVIEW

File: test/unit/au/org/ala/regions/SpeciesIpControllerSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import grails.test.mixin.TestFor
+import spock.lang.Specification
+
+/**
+ * See the API for {@link grails.test.mixin.web.ControllerUnitTestMixin} for usage instructions
+ */
+@TestFor(SpeciesIpController)
+class SpeciesIpControllerSpec extends Specification {
+
```
Comment (aruizca) [inline test/unit/au/org/ala/regions/SpeciesIpControllerSpec.groovy]: Remove this from version control if no actual test

File: test/unit/au/org/ala/regions/SpeciesIpServiceSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package au.org.ala.regions
+
+import grails.test.mixin.TestFor
+import spock.lang.Specification
+
+/**
+ * See the API for {@link grails.test.mixin.services.ServiceUnitTestMixin} for usage instructions
+ */
+@TestFor(SpeciesIpService)
+class SpeciesIpServiceSpec extends Specification {
```
Comment (aruizca) [inline test/unit/au/org/ala/regions/SpeciesIpServiceSpec.groovy]: Remove this from version control if no actual test

File: grails-app/conf/UrlMappings.groovy [hunk 3]
```diff
@@ -4,6 +4,13 @@ class UrlMappings {
 
         "/patents" controller: "patent", action: [POST: "patents"]
 
+        "/$regionType/$regionName/patents" (controller: 'speciesIp', action: 'region') {
```
Comment (aruizca) [inline grails-app/conf/UrlMappings.groovy:7]: Why do we need need this webservice? I thought there would be a batch job that automatically fetches the data from auspat. How I am supposed to use this when deployed?

**Comment by charvolant** [comment]
The service is for Temi to get patent data based on the region

File: test/unit/au/org/ala/regions/SpeciesIpServiceSpec.groovy (hunk 2 above)
Comment (charvolant) [inline test/unit/au/org/ala/regions/SpeciesIpServiceSpec.groovy]: Removed

**Pull request merged**